### PR TITLE
fix(onboarding): warnings for external host errors

### LINKED ISSUE #15101: Suppress external host errors when onboarding
### What would you like Renovate to be able to do?

Create an onboarding PR even if external host errors are encountered during lookup

### If you have any ideas on how this should be implemented, please tell us here.

We have a few hosts where we throw an external host error (e.g. Docker Hub) and this results in the run being aborted. Instead, we should continue and display a warning in the onboarding PR so they can fix it using config.

### Is this a feature you are interested in implementing yourself?

No

### PR DESCRIPTION
**Pull request by zharinov**
## Changes

- Catch errors thrown in `lookupUpdates`
- Produce warning messages for `ExternalHostError` instances, otherwise re-throw

## Context

- Closes: #15101

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by zharinov** [comment]
Example: https://github.com/renovate-testing/test-15101-onboarding-external-errors/pull/2

File: lib/workers/repository/process/fetch.ts [hunk 1]
```diff
@@ -42,11 +44,25 @@ async function fetchDepUpdates(
...
-        ...dep,
-        ...(await lookupUpdates(depConfig as LookupUpdateConfig)),
-      };
+      try {
+        dep = {
+          ...dep,
+          ...(await lookupUpdates(depConfig as LookupUpdateConfig)),
+        };
+      } catch (err) {
+        if (config.repoIsOnboarded || !(err instanceof ExternalHostError)) {
```
Comment (viceice) [inline lib/workers/repository/process/fetch.ts:53]: i think you can use `packageFileConfig` here instead?

Comment (zharinov) [reply lib/workers/repository/process/fetch.ts]: Good point, thanks 👍

**Comment by rarkins** [comment]
The error displayed is pretty hard to understand! Different problem of course but we should fix it next

**Review by rarkins — APPROVED** [review]

**Pull request merged**